packages/utils/src/lib/zod-validation.unit.test.ts

```diff
@@ -0,0 +1,14 @@
+import { formatErrorPath } from './zod-validation';
+
+describe('formatErrorPath', () => {
+  it.each([
+    [['categories', 1, 'slug'], 'categories[1].slug'],
+    [['plugins', 2, 'groups', 0, 'refs'], 'plugins[2].groups[0].refs'],
+    [['refs', 0, 'slug'], 'refs[0].slug'],
+    [['categories'], 'categories'],
+    [[], ''],
+    [['path', 5], 'path[5]'],
+  ])('formats error path correctly for $input', (input, expected) => {
```

Comment:
The `$`-syntax only works when the test case is an object. But since you have tuples, `$input` isn't replaced and there are duplicate test names:

![image](https://github.com/user-attachments/assets/f3b8587d-27ad-479b-98cc-2076eeb4f6ea)

The `%`-syntax should be used when your cases are defined as tuples:

```suggestion
  ])('formats error path correctly for %j', (input, expected) => {
```

![image](https://github.com/user-attachments/assets/48b14b5c-8e59-4ec8-b4bd-10c8343051f7)

Although it may be even nicer to include the expected string in the test name:

```suggestion
  ])('should format error path %j as %j', (input, expected) => {
```

![image](https://github.com/user-attachments/assets/30d34840-221d-43d1-af0f-f2f5ea10186c)

(Tip: To expand all the tests in the Vitest output, append `-- --reporter=verbose` to your command.)

Reply:
Thanks for the tip! Didn't know that.